src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystem.java

```diff
@@ -96,13 +96,17 @@ private static void createLocalVoltageControlEquation(LfBus bus, LfNetworkParame
         }
     }
 
-    private static void createReactivePowerControlBranchEquation(LfBranch branch, ReactivePowerControl.ControlledSide controlledSide,
-                                                                 EquationSystem<AcVariableType, AcEquationType> equationSystem, EquationTerm<AcVariableType, AcEquationType> q) {
-        branch.getReactivePowerControl().ifPresent(reactivePowerControl -> {
-            if (reactivePowerControl.getControlledSide() == controlledSide) {
-                equationSystem.createEquation(branch.getNum(), AcEquationType.BRANCH_TARGET_Q).addTerm(q);
-            }
-        });
+    private static void createReactivePowerControlBranchEquation(LfBranch branch, LfBus bus1, LfBus bus2, EquationSystem<AcVariableType, AcEquationType> equationSystem,
+                                                                 boolean deriveA1, boolean deriveR1) {
+        if (bus1 != null && bus2 != null) {
```

Comment:
I understand your fix, but I wonder if our building or `ReactivePowerControl` is safe enough...

Reply:
Yes, we should add more unit test to test different network conditions. Here is we should at least add a test with an open branch and maybe also test the lost of controlled branch in the security analysis.

Reply:
In this PR?

Reply:
Non let's keep it as it is in this PR and create another one later.